### ISSUE TITLE
Implement scheduled task in order to delete 

### ISSUE THREAD

**Issue by dasistwas** [dasistwas/moodle-mod_newsletter#34]
I think the table newsletter_deliveries is only used for tracking if a mail was delivered in order to not forget to send the mail to users when the cron jobs interrupts or the server shuts down. So the sending of mails will restart with the first user not yet processed. 
This db contains a lot of entries so there should be a scheduled task that checks if the whole newsletter was already delivered (table newsletter_issues field delivered = 1) and if that is the case, delete all entries for that newsletter issue in table newsletter_deliveries). Check if that table is used for any other purpose before starting ;-)

**Comment by michaelpollak** [comment]
It is also used to count how many mails were sent and how many remain to be sent. I assume this feature is not too important, we can add a second timer, for example we could remove all details after sending is completed and 4 weeks after publishing. 
Will start implementing this.

**Comment by dasistwas** [comment]
OK, wait: We have to do this:

1) For sending newsletter this has to stay in place
2) After alle mails have been sent the newsletter_issues table field "delivered" is set to 1
3) So scheduled task has to check if delivered = 1 and only then execute the task
4) Before deleting, the number of sent mails has to be saved elsewhere: In newsletter_subscriptions there is the field sentnewsletters, there is a good place to store the value (increase by 1). I think it is already done. Yes, just checked.

**Comment by michaelpollak** [comment]
Just commited my solution, I will check you #4, rest should be as discussed.

**Comment by michaelpollak** [comment]
In sentnewsletters we store the complete number of sent mails, what we are missing after clearing deliveries is the number teachers see for every issue. There is a blue bar as long as not all deliveries are sent, once sending is completed you get a blue 100% full bar with the number of mails delivered of this issue. This information is read from the deliveries table and can not be found elsewhere. If it is important we can add another database entry but I think just marking it as completed should be just fine, see code.

**Comment by dasistwas** [comment]
OK, maybe it is not a good idea to delete these entries. Let us just leave the newsletter tool and go back to datalynx.
